File: src/2401. Longest Nice Subarray.cpp
class Solution {
public:
    int longestNiceSubarray(vector<int>& nums) {
        int j = 0, used = 0, res = 0;
        for (int i = 0; i < nums.size(); ++i) {
            while (used & nums[i]) {
                used ^= nums[j];
                j++;
            }
            used |= nums[i];
            res = max(res, i - j + 1);
        }
        return res;
    }
};

File: src/2401. Longest Nice Subarray.py
class Solution:
    def longestNiceSubarray(self, nums: List[int]) -> int:
        j = used = res = 0
        for i in range(len(nums)):
            while used & nums[i]:
                used ^= nums[j]
                j += 1
            used |= nums[i]
            res = max(res, i - j + 1)
        return res